src/app/features/reporting/reporting/reporting.component.ts

```diff
@@ -34,8 +31,8 @@ import { ReportConfig } from "../report-config";
   standalone: true,
 })
 export class ReportingComponent {
-  reports: ReportConfig[];
-  mode: "exporting" | "reporting" = "exporting";
+  reports: ReportType[];
+  mode: ReportType["mode"];
```

Comment:
I find `ReportType` a bit confusing, as this here is more than just the type. From the names I would always intuitively write ReportConfig instead of ReportType. Can't think of a better naming right now, however ...

Alternatively, maybe work with a generic on ReportConfig instead of the sup-classes? Then we wouldn't have 1 + 3 + 1 different types in this context

Reply:
Lets see if can find a better approach here